### PR TITLE
Upgrade PHPUnit to v8

### PR DESCRIPTION
**Pull request by MarioBlazek**
This PR upgrades PHPUnit to v8 and fixes deprecations along the way.

### CODE REVIEW

File: tests/Monolog/Formatter/GelfMessageFormatterTest.php [hunk 1]
```diff
@@ -82,10 +83,11 @@ public function testFormatWithFileAndLine()
 
     /**
      * @covers Monolog\Formatter\GelfMessageFormatter::format
-     * @expectedException InvalidArgumentException
      */
     public function testFormatInvalidFails()
     {
+        $this->expectException(\InvalidArgumentException::class);
```
Comment (gmponos) [inline tests/Monolog/Formatter/GelfMessageFormatterTest.php]: The reason the annotations are deprecated is in order to explicitly be added before the class that will throw the exception.. So this must be moved below.. and excactly above the function that will throw the exception

Comment (gmponos) [reply tests/Monolog/Formatter/GelfMessageFormatterTest.php]: Same thing for other places as well.

Comment (MarioBlazek) [reply tests/Monolog/Formatter/GelfMessageFormatterTest.php]: @gmponos thanks for the review. I will update PR accordingly.

**Comment by Seldaek** [comment]
Thanks

**Pull request merged**